specification/quota/resource-manager/Microsoft.Quota/preview/2021-03-15-preview/quota.json

```diff
@@ -0,0 +1,1226 @@
+          "type": "string",
+          "description": "Resource type"
+        },
+        "value": {
+          "description": "Quota type information.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/QuotaTypeDimensionInformation"
+          }
+        }
+      }
+    },
+    "QuotaTypeDimensionInformation": {
+      "type": "object",
+      "description": "Quota type information.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Property name."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Display name."
+        },
+        "id": {
+          "type": "string",
+          "description": "Dimension ID."
+        }
+      }
+    },
+    "QuotaTemplateDetails": {
+      "type": "object",
+      "description": "Quota template details.",
+      "properties": {
+        "resourceType": {
+          "type": "string",
+          "description": "Resource type."
+        },
+        "properties": {
+          "description": "Quota type properties for the resource ID.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/QuotaTypeInformationDimensionList"
+        }
+      }
+    },
+    "resourceQueryDetails": {
+      "type": "object",
+      "description": "Resource query details.",
+      "properties": {
+        "resourceQueryType": {
+          "description": "Resource query type.",
+          "$ref": "#/definitions/ResourceQueryType"
+        },
+        "resourceQueryMethod": {
+          "description": "Resource query method.",
+          "$ref": "#/definitions/ResourceQueryMethod"
+        },
+        "resourceQueryUri": {
+          "type": "string",
+          "description": "Base URI for for resource query."
+        },
+        "resourceQueryPostTemplate": {
+          "description": "Template to create the resource query.",
+          "type": "string"
+        }
+      }
+    },
+    "ResourceQueryType": {
+      "description": "Resource query types. For extensibility, it is a string.",
+      "enum": [
+        "ARG",
+        "RestAPI"
+      ],
+      "readOnly": false,
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ResourceQueryType",
+        "modelAsString": true
+      }
+    },
+    "ResourceQueryMethod": {
+      "description": "The resource query method.",
+      "enum": [
+        "GET",
+        "POST"
+      ],
+      "readOnly": false,
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ResourceQueryMethod",
+        "modelAsString": true
+      }
+    },
+    "QuotaTemplatesDetails": {
+      "type": "object",
+      "description": "Quota templates details.",
+      "properties": {
+        "value": {
+          "description": "Quota templates information.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/QuotaTemplateDetails"
+          }
+        }
+      }
+    },
+    "ServiceError": {
```

Comment:
>ServiceError [](http://example.com/codeflow?start=5&length=12)

These definitions exist in common types v2 and you could reuse them. (This is non-blocking)

Reply:
I will check it out.